src/mcs/mcs_types.jl

```diff
@@ -183,15 +185,25 @@ mutable struct SimulationInstance{T}
     models::Vector{M} where M <: AbstractModel
     results::Vector{Dict{Tuple, DataFrame}}
     payload::Any
+    payload2::Any
     translist_modelparams::Vector{TransformSpec_ModelParams} 
 
     function SimulationInstance{T}(sim_def::SimulationDef{T}) where T <: AbstractSimulationData
         self = new()
         self.trials = 0
         self.current_trial = 0
         self.current_data = nothing
+
+        # This is a bit convoluted, but we want to make sure to exclude payload2 from the deepcopy
+        # call here
+        payload2 = sim_def.payload2
+        sim_def.payload2 = nothing
         self.sim_def = deepcopy(sim_def)
```

Comment:
Just so I'm clear here, what we're doing is (1) deep copy the `sim_def` without payloads  (2) add the `payload` and the `payload2` (3) replace the `payload` in the `sim_def`  with a deep copied version